File: LeetCode2-Math/CountPrimes/solution.py
class Solution:
    def countPrimes(self, n: int) -> int:
        
        if n is 0:  
            return 0

        isPrime = [True]*n

        for i in range(2,n):
            if isPrime[i]:
                for j in range(i*2, n, i):
                    isPrime[j] = False
                

        count = 0

        for i in range(2,n):
            if isPrime[i]:
                count+=1

        return count
        

File: LeetCode2-Math/CountPrimes/solution.cpp
class Solution {
public:
    int countPrimes(int n) {

        if (n==0) return 0;

        vector<bool> isPrime(n, true);

        for (int i = 2; i*i < n; i++){

            if (isPrime[i]){
                for (int j = i*2; j < n; j+= i){
                    isPrime[j] = false;
                }
            }

        }

        int count = 0;

        for (int i = 2; i < n; i++){
            if (isPrime[i]) count++;
        }

        return count;
        
    }
};